pilot/pkg/config/kube/gateway/conditions.go

```diff
@@ -0,0 +1,206 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gateway
+
+import (
+	"sort"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "sigs.k8s.io/gateway-api/apis/v1alpha1"
+
+	"istio.io/istio/pilot/pkg/model/kstatus"
+	"istio.io/istio/pkg/config"
+	"istio.io/istio/pkg/config/constants"
+)
+
+func createRouteStatus(gateways []string, obj config.Config, current []k8s.RouteGatewayStatus, routeErr *ConfigError) []k8s.RouteGatewayStatus {
+	setGateways := map[localNamedReference]struct{}{}
+	for _, gw := range gateways {
+		ref := localNamedReference{}
+		if gw == constants.IstioMeshGateway {
+			ref.Name = experimentalMeshGatewayName
+			// TODO this is not namespaced but a namespace is required
+			ref.Namespace = "default"
+		} else {
+			s := strings.Split(gw, "/")
+			ref.Name = s[1]
+			ref.Namespace = s[0]
+		}
+		setGateways[ref] = struct{}{}
+	}
+	gws := make([]k8s.RouteGatewayStatus, 0, len(gateways))
+	// Fill in all of the gateways that are already present but not owned by use
+	for _, r := range current {
+		if r.GatewayRef.Controller == nil {
+			// Controller not set. This may be our own resource due to using old CRDs that prune the controller field
+			_, f := setGateways[localNamedReference{r.GatewayRef.Name, r.GatewayRef.Namespace}]
+			if !f {
```

Comment:
for my own understanding, this says that if we own this gateway (based on the `setGateways`), we don't append to the result because expect to rebuild this status from `gateways`?

Reply:
yep. I tried to make the comments a bit more clear